.github/workflows/code-analysis.yml

```diff
@@ -0,0 +1,24 @@
+name: PHP Code Analysis
+
+on:
+  pull_request:
+    branches: [ main ]
+  push:
+    branches: [ main ]
+
+jobs:
+  php-stan:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Pull repo
+      uses: actions/checkout@v2
+    - run: composer install
+    - run: sudo rm phpstan.neon.dist 
+    - run: sudo touch phpstan.neon.dist 
+    - run: sudo chown -R $USER:$USER phpstan.neon.dist
+    - run: sudo echo 'parameters:' >> phpstan.neon.dist
+    - run: sudo echo '    paths:' >> phpstan.neon.dist
+    - run: sudo echo '        - %currentWorkingDirectory%/' >> phpstan.neon.dist
+    - run: sudo echo '    excludes_analyse:' >> phpstan.neon.dist
+    - run: sudo echo '        - %currentWorkingDirectory%/vendor/*' >> phpstan.neon.dist
+    - run: vendor/bin/phpstan analyze --level max
```

Comment:
Probably best to configure the level via phpstan.neon so that every one runs it at the same level.

Reply:
now defined level in phpstan.neon.dist file